python_script/meter-health/meter_health.py

```diff
@@ -0,0 +1,119 @@
+import requests
+import csv
+import logging
+logging.basicConfig(filename='example.log',level=logging.DEBUG)
+from oauthlib.oauth2 import LegacyApplicationClient
+from requests_oauthlib import OAuth2Session
+from config.credentials import *
+
+
+class MeterHealth:
+    # URL information for the API. ONLY change this method if you know what you are doing
+    BASE_URL="http://54.90.173.204:8080"
+    ORGANIZATIONS="/api/v2.json"
+    METERS="/api/v2/organizations/%d/meters.json"
+    TOKEN_URL="/oauth/token"
+
+    def __init__(self):
+        self.token=self.retrieve_token()
+        self.payload={'access_token': self.token}
+
+    def create_report(self):
+        """ It is the main method that calls
+        all that need to be done
+        """
+        if self.token:
+            organizations=self.retrieve_organizations()
+            if organizations:
+                self.build_csv(organizations)
+            else:
+                print("There aren't organizations \n")
+
+    def build_csv(self, organizations):
+        # Creates/Opens the file where the info will be created
+        total_org=len(organizations)
+        print("We found %d organizations \n" % total_org)
+        with open(file_name, 'w+') as csv_file:
+            writer=csv.writer(csv_file)
+            writer.writerow(['Organization Name', 'Meter ID', 'Meter Name',
+                             'Meter Type', 'Meter Status', 'Last Reading'])
+            self.add_information_to_csv({'organizations': organizations,'writer': writer, 'total_org': total_org})
+            print("Process successfully finished, analyzed a total of %d organizations" % total_org)
+
+    def add_information_to_csv(self, params):
+        # Iterates over the organizations to retrieve meters
+        organization_index=1
+        for organization in params['organizations']:
+            print("Retrieving meters for organization %d of %d ...\n" % (organization_index, params['total_org']))
+            organization_index += 1
+            meter_info=self.retrieve_meter_info_for(organization['remote_id'])
+            self.write_meters_to_csv({"meter_info": meter_info,"organization": organization,"writer": params["writer"]})
+
+    def write_meters_to_csv(self, params):
+        """ Iterates over a meters for an organization
+        to write the info to csv
+        """
+        for meter in params['meter_info']:
+            params['writer'].writerow([params['organization']['name'], meter['remote_id'], meter['name'],
```

Comment:
Since there are 4 parameters passed to that function, to improve readability you could do something like:
```
...writerow(param1,
            param2,
            param3,
            param4)
```